src/clang-cpp-frontend/clang_cpp_adjust_expr.cpp

```diff
@@ -152,15 +152,7 @@ void clang_cpp_adjust::adjust_cpp_member(member_exprt &expr)
 
 void clang_cpp_adjust::adjust_if(exprt &expr)
 {
-  // Check all operands
-  adjust_operands(expr);
-
-  // If the condition is not of boolean type, it must be casted
-  gen_typecast(ns, expr.op0(), bool_type());
-
-  // Typecast both the true and false results
-  gen_typecast(ns, expr.op1(), expr.type());
-  gen_typecast(ns, expr.op2(), expr.type());
+  clang_c_adjust::adjust_if(expr);
```

Comment:
if this method is simply calling the overridden method, you can remove it from clang_cpp_adjust